### PR TITLE
clean up E0436 explanation

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @Dylan-DPC

### CODE REVIEW

File: src/librustc_error_codes/error_codes/E0436.md [hunk 1]
```diff
@@ -1,5 +1,4 @@
-The functional record update syntax is only allowed for structs. (Struct-like
-enum variants don't qualify, for example.)
+The functional record update syntax was used on something else than a struct.
```
Comment (Dylan-DPC) [inline src/librustc_error_codes/error_codes/E0436.md]:
```suggestion
The functional record update syntax was used on something other than a struct.
```

**Comment by GuillaumeGomez** [comment]
Updated.

**Pull request merged**